content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1066 @@
+
+Also note that you can rename [relation fields](/concepts/components/prisma-schema/relations#relation-fields) to optimize the Prisma Client API that you'll use later to send queries to your database. For example, the `post` field on the `user` model is a _list_, so a better name for this field would be `posts` to indicate that it's plural.
+
+Update the `published` field by including the `@default` attribute to define the default value of the field.
+
+You can also rename the `UserProfile` composite type to `Profile`.
+
+Here's an adjusted version of the Prisma schema that addresses these points:
+
+```prisma file=prisma/schema.prisma highlight=10,14,17,22,25,29,30,38,41,43,46,49;normal
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+type Profile {
+  bio String
+}
+
+model Category {
+  id       String @id @default(auto()) @map("_id") @db.ObjectId
+  name     String
+  v        Int    @default(0) @map("__v") @ignore
+
+  posts    Post[]   @relation(fields: [post_ids], references: [id])
+  post_ids String[] @db.ObjectId
+
+  @@map("categories")
+}
+
+model Post {
+  id           String  @id @default(auto()) @map("_id") @db.ObjectId
+  title        String
+  content      String
+  published    Boolean @default(false)
+  v            Int    @default(0) @map("__v") @ignore
+
+  author       User   @relation(fields: [authorId], references: [id])
+  authorId     String @map("author") @db.ObjectId
+
+  categories   Category[] @relation(fields: [categoryIds], references: [id])
+  categoryIds  String[]   @db.ObjectId
+
+  @@map("posts")
+}
+
+model User {
+  id      String        @id @default(auto()) @map("_id") @db.ObjectId
+  v       Int           @default(0) @map("__v") @ignore
+  email   String        @unique(map: "email_1")
+  name    String
+  profile Profile?
+  posts   Post[]
+
+  @@map("users")
+}
+```
+
+## Step 3. Install Prisma Client
+
+As a next step, you can install Prisma Client in your project so that you can start replacing the database queries in your project that are currently made with Mongoose:
+
+```terminal
+npm install @prisma/client
+```
+
+## Step 4. Replace your Mongoose queries with Prisma Client
+
+In this section, we'll show a few sample queries that are being migrated from Mongoose to Prisma Client based on the example routes from the sample REST API project. For a comprehensive overview of how the Prisma Client API differs from Mongoose, check out the [API comparison](/concepts/more/comparisons/prisma-and-mongoose#api-comparison) page.
+
+First, to set up the `PrismaClient` instance that you'll use to send database queries from the various route handlers. Create a new file named `prisma.js` in the `src` directory:
+
+```terminal copy
+touch src/prisma.js
+```
+
+Now, instantiate `PrismaClient` and export it from the file so you can use it in your route handlers later:
```

Comment:
Alex, the phrasing "... so that you can export it from the file..." is a bit confusing. What is "it" in this situation?

If "it" is the contents of the `prisma.js` file, then maybe rewording to something like this would help:

" Now, instantiate `PrismaClient` and export the contents from the `prisma.ts` file in order to create a new Client that you can use in your route handlers later:"

Reply:
The "it" refers to the `PrismaClient` instance. I don't think there will be a need to reword it.

Reply:
@ruheni on this one, I'm OK with not spelling out what "it" is, but we really do need to be more clear about how to export it.... i.e. which file? So let's add in the file name (prisma.ts). It's always good to specify file names; we have a quite a few and shouldn't assume that al users know what is in each one.

Reply:
In this case, we're exporting an instance of Prisma Client which will be contained in the `prisma.js` file. The file isn't exported here.